webgpu-native/webgpu-headers
# generate header from yaml spec

Add a custom yaml spec as discussed in #207, the details about the format can be found in the `schema.json`.

To test the generator:
1. First make sure you have Go installed.
1. Make some change in `webgpu.yml`
1. Run `make gen`

The generator can also be used to generate a separate implementation specific C header along side `webgpu.h`. The steps to do that is:
1. Create a new yaml file, for example `wgpu.yml`
1. Populate it with minimal required fields, refer to `schema.json` and [this example](https://gist.github.com/rajveermalviya/15200babba779bfc21d837af504a1860).
1. Run `go run ./gen -schema schema.json -yaml wgpu.yml -header wgpu.h`

Another option is to generate a single header from multiple yaml specs, which is also included but disabled currently.
I prefer separate headers because that's what `wgpu-native` uses currently.
